frontend/js/components/router/router.js

```diff
@@ -60,9 +60,22 @@ export class Router extends HTMLElement {
   static async #addViewToRouterDom (route) {
     const { view, prescript } = Router.#routes[route]
     Router.#root.innerHTML = await (await fetch(`/views/${view}.html`)).text()
+
     if (prescript) {
-      const [modulePath, func] = prescript.split('#')
-      import(`../../${modulePath}.js`).then(mod => mod[func]())
+      this.importFunc(prescript)
+    }
+
+    const conditions = Router.#root.querySelectorAll('[renderIf]')
+    for (const condition of conditions) {
+      if (!await this.importFunc(condition.attributes.renderIf.value)) {
+        condition.remove()
+      }
     }
```

Comment:
Did not understand this part (lines 68-72). Also, how does !await (await with negation) operator works?

Reply:
Await with negation works absolutely same way as with other functions, you can think of it as **_!(await func())_**, it will just negate the return result. Just imagine _**func()**_ returns **_Promise.resolve(2 === 2)_**. In this case **_await func()_** returns true and negation just changes it to false.
Regarding to this block, we check every html element with renderIf tag. If the return result of linked to this tag function is false then we delete this element from the page.